### DISCUSSION TITLE
failed to open terminal: ENOTTY: Not a typewriter

### DISCUSSION THREAD

**Discussion by papanito** [zellij-org/zellij#1690] [Q&A] [answered]
When I start a terminal (Gnome Terminal, Tilix), I get the following error
```
failed to open terminal:
ENOTTY: Not a typewriter
```
My default terminal is `zsh` and this is in my `.zshrc` config

```
eval "$(zellij setup --generate-auto-start zsh)"
```

I have seen [this issue](https://github.com/zellij-org/zellij/issues/1534) but it's about fish.

In case, here is my [.zshrc](https://github.com/papanito/dotfiles/blob/main/dot_zshrc)

**Comment by jaeheonji** [comment]
Hi, @papanito!

I did the test, but this is not reproduced for me. Note that my test environment is `zsh 5.8.1` and `zellij 0.31.3`.
And I put only the setup script in `.zshrc`.

Could you try to put raw script in `.zshrc` like below?
```zsh
...

if [[ -z "$ZELLIJ" ]]; then
    if [[ "$ZELLIJ_AUTO_ATTACH" == "true" ]]; then
        zellij attach -c
    else
        zellij
    fi

    if [[ "$ZELLIJ_AUTO_EXIT" == "true" ]]; then
        exit
    fi
fi
```

**Reply by papanito** [reply]
Thanks for looking into it. Yeah I tried that, but still get the same error. If I reload the `.zshrc`-config, the issue does not appear. Really only when opening a new terminal

**Reply by papanito** [reply]
I added some echo statements to see until where it goes, and actually it goes to the line where `zellij` is called

**Reply by jaeheonji** [reply]
This is very weird. It seems that this only doesn't work when opening a new window in a specific terminal emulator. Because, when I tested it on `wezterm` and `alacritty` there was no problem.

I'll try some more tests to find the problem. Could you please tell me your OS and version of Gnome Terminal to configure the same environment as yours?

**Reply by papanito** [reply]
Interesting, I also have the issue when I open one of the `tty` e.g. by pressing `Ctrl+Alt+F3`. I am using Arch linux 5.19.3-arch1-1, while my standard terminal is

```
zsh --version
zsh 5.9 (x86_64-pc-linux-gnu)
```

Other versions

```
gnome-terminal --version
# GNOME Terminal 3.44.1 using VTE 0.68.0 +BIDI +GNUTLS +ICU +SYSTEMD
```
or

```
tilix -v
	Tilix version: 1.9.5
	VTE version: 0.68
	GTK Version: 3.24.34
```

**Reply by papanito** [reply]
It basically happens with every terminal, also tried with `wezterm`,

**Reply by papanito** [reply]
I suspect it may be related to https://github.com/romkatv/powerlevel10k#instant-prompt. Will dig further....

**Reply by azzamsa** [reply]
@papanito do you find the solutions? I have the same issue both in gnome terminal and wezterm. I also use `p10k` instant prompt. What's your current workaround?

**Reply by papanito** [reply]
I did not follow up on it for now, I just disabled autostart for now and start it manually

**Reply by arttatum** [reply, marked as answer]
I faced the same issue (Kitty, p10k, zsh, zellij)

I worked around the issue by evaluating the zellij autostart command before sourcing p10k.

**Reply by papanito** [reply]
> I worked around the issue by evaluating the zellij autostart command before sourcing p10k.

I can confirm, this worked for me too
